### ISSUE TITLE
Feature request: maintain rockspec  with multiple users

### ISSUE THREAD

**Issue by kenhys** [luarocks/luarocks-site#127]
## Problem

It seems that there is no way to maintain rockspec with multiple users.

## Actual

There is one solution to maintain rockspec with multiple users - share one account with among them.
But I hope to maintain rockspec with multiple users - maintain rockspec by group.

## Expected

Allow to join specific group and maintain rockspec with multiple users. (I assume that owner and permitted users  to upload rockspec)
When some person can't maintain specific rockspec anymore, others can do if rockspec is maintained by group.

**Comment by Alloyed** [comment]
yeah; I am finding myself in the same situation, where I would like to my projects that have joint maintainership the ability to upload their own versions of existing packages under my account.

here are some approaches I've been able to pull out of my brain, with more/less effort involved on either the repository maintainer or the luarocks side:

1. Create API keys, then on the luarocks website restrict these API keys to only function for a given module. requires some work on the luarocks end, but then you can just hand a string to any given maintainer. 
2. New account per module: no feature work, but a bit of a pain. if a package is already listed under a specific account, switching accounts can break links to it.
3. Write a bot that auto-picks-up rockspecs pushed to the repository and has an API key associated with my account. then anybody with write access or the ability to make a PR can make rockspec changes, but requires somebody to actually write that bot. may be a good way to (misuse) github actions.
